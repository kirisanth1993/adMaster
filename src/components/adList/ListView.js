import './ListView.scss';
import { Grid, Button, Box } from '@material-ui/core';
import React, { useEffect, useState } from 'react';
import DropDown from '../common/DropDown';
import { useDispatch, useSelector } from "react-redux";
import SingleAdvertisement from './SingleAdvertisement';
import { ChangeAdvertiseList } from '../../actions/ChangeAdvertiseList';
import { ChangeCategory } from '../../actions/ChangeCategory';

const ListView = (props) => {
    // redux saving action
    const dispatch = useDispatch();
    // get data from redux store
    const {categoryList} = useSelector((state) => state);
    const {advertisementList} = useSelector((state) => state);
    const {category} = useSelector((state) => state);    
    const [ selectedCategory, setSelectedCategory ] = useState("all");
    const [ categoryOptions, setCategoryOptions ] = useState(categoryList);
    const [ displayAdvertisementList, setDisplayAdvertisementList ] = useState([]);

    const categorySelectionAction = (event) => {
        setSelectedCategory(event.target.value);
        dispatch(ChangeCategory(event.target.value));
    }

    useEffect(() => {
        setSelectedCategory(category);
    },[]);

    // filtering action
    useEffect(() => {
        let filteredAdList = advertisementList.filter((singleAd) => {
            if((selectedCategory === "all") || (selectedCategory === singleAd.category)){
                if( ((props.currentPageId === 1) && singleAd.isFav) || (props.currentPageId === 0)){
                    return(singleAd);
                }
            }
        });
        setDisplayAdvertisementList(filteredAdList);
    },[advertisementList, props.currentPageId, selectedCategory]);

    const favChangeAction = (value, item) => {
        advertisementList.map((singleAd) => {
            if(item.id === singleAd.id){
                singleAd.isFav = value;
            }
        });
        dispatch(ChangeAdvertiseList([...advertisementList]));
    }

    return(
        <Grid>
            {/* category selection and new btn */}
            <Grid container>
                <Grid item xs={2}>
                    <DropDown
                        id="Category"
                        label="Category"
                        selectedCategory={ selectedCategory }
                        categorySelectionAction={ (e) => { categorySelectionAction(e); } }
                        options={ categoryOptions }
                    />
                </Grid>
                <Grid item xs={10} container justifyContent={"flex-end"}>
                    <Box display="flex" justifyContent="flex-end">
                        <Button 
                            variant="contained" 
                            color="primary" 
                            className="new-btn"
                            onClick={ () => props.newBtnAction() }
                            >
                            New Classified
                        </Button>
                    </Box>
                </Grid>
            </Grid>

            {/* list view */}
            <Grid container className="advertisement-list-block" spacing={3} display="flex" justifyContent="space-between">
                {
                    displayAdvertisementList &&
                    displayAdvertisementList.map((SingleAdvertisementData, SingleAdvertisementDataIndex) => {
                        return(
                            <Grid item className="singleBlock" key={ SingleAdvertisementDataIndex }>
                                {/*  single component for each ads */}
                                <SingleAdvertisement 
                                    adDetail={ SingleAdvertisementData }
                                    checkboxChange={ (value, item) => favChangeAction(value, item) }
                                    itemClickAction={ (ad) => { props.adClickAction(ad) } }
                                />
                            </Grid>
                        );
                    })
                }
                {
                    // error message
                   displayAdvertisementList && displayAdvertisementList.length === 0 &&
                   <Grid container display="flex" justifyContent="center" className="nodata">No Data Found</Grid>
                }
                

            </Grid>

        </Grid>

    );
}
export default ListView;