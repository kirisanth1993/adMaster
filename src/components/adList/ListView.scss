.new-btn{
    border-radius: 50px !important;
}

.advertisement-list-block{
    margin-top: 25px !important;
}

.single-ad-container{
//    height: 365px;
}

.singleBlock{
    width: 27%;
}

.single-ad-img{
    width: 100%;
    height: 168px;
    border-radius: 5px 5px 0px 0px;
}

.info-page-single-ad-img{
    height: 320px !important;
}

.single-ad-inner-block{
    padding: 10px 20px;
}

.single-ad-category{
    text-transform: uppercase;
    font-weight: 500;
    font-size: 10px;
}

.single-ad-title{
    font-weight: 500;
    font-size: 20px;
    margin-top: 5px;
    margin-bottom: 5px;
}

.info-page-single-ad-title{
    font-size: 34px !important;
}

.single-ad-description{
    font-weight: 400;
    font-size: 14px;
    margin-bottom: 10px;
}

.info-page-single-ad-description{
    font-size: 18px !important;
}

.fav-checkbox{
    position: absolute;
    top: 10px;
    right: 10px;
    background-color: white;
    border-radius: 50%;
}

.fav-img-wrapper{
    position: relative;
}

.nodata{
    color: red;
    font-size: 50px;
    margin-top: 40px;
}