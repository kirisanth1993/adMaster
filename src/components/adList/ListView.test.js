import React from 'react';
import ReactDOM from 'react-dom';
import ListView from './ListView';
import { Provider } from "react-redux";
import { cleanup } from '@testing-library/react'
import "@testing-library/jest-dom/extend-expect"
import store from "../../store";

import renderer from "react-test-renderer";

afterEach(cleanup);

it('renders without crashing', () => {
    const div = document.createElement("div");
    ReactDOM.render(<Provider store={store}><ListView></ListView></Provider>, div);
    ReactDOM.unmountComponentAtNode(div);
});


it("it matches snapshot", () => {
    const tree = renderer.create(<Provider store={store}><ListView></ListView></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
})