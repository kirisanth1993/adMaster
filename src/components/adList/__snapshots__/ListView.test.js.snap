// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<div
  className="MuiGrid-root"
>
  <div
    className="MuiGrid-root MuiGrid-container"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
    >
      <div
        className="MuiFormControl-root MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
          data-shrink={true}
          id="demo-simple-select-outlined-label"
        >
          Category
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          onClick={[Function]}
        >
          <div
            aria-haspopup="listbox"
            aria-labelledby="Category"
            className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
            id="Category"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="button"
            tabIndex={0}
          >
            All
          </div>
          <input
            aria-hidden={true}
            className="MuiSelect-nativeInput"
            onAnimationStart={[Function]}
            onChange={[Function]}
            required={false}
            tabIndex={-1}
            value="all"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-8 PrivateNotchedOutline-legendNotched-9"
            >
              <span>
                Category
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-content-xs-flex-end MuiGrid-grid-xs-10"
    >
      <div
        className="MuiBox-root MuiBox-root-10"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained new-btn MuiButton-containedPrimary"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            New Classified
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root advertisement-list-block MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-justify-content-xs-space-between"
    display="flex"
  >
    <div
      className="MuiGrid-root nodata MuiGrid-container MuiGrid-justify-content-xs-center"
      display="flex"
    >
      No Data Found
    </div>
  </div>
</div>
`;
